### PR TITLE
Enhance Release Event Card for Mobile View closes #358

### LINKED ISSUE #358: Enhance Release Event Card for Mobile View
https://contributors.ohc.network/releases

image

### PR DESCRIPTION
**Pull request by rash-27**
- closes #358 
- if width is above 390px the care/org is shown
Screenshot from 2024-03-11 01-10-01
- if width is below 390px it is hidden
Screenshot from 2024-03-11 01-08-52

### CODE REVIEW

File: app/releases/page.tsx [hunk 1]
```diff
@@ -50,19 +50,25 @@ export default async function Page() {
...
+                  className="text-sm font-semibold sm:text-base"
                 >
                   {release.author.login}
                 </a>{" "}
+                on{" "}
+                {new Date(release.createdAt).toLocaleDateString("en-US", {
+                  day: "2-digit",
+                  month: "short",
+                  year: "numeric",
+                })}
```
Comment (rithviknishad) [inline app/releases/page.tsx:67]: Use the custom `formatDate` utility function instead, so that we have consistent date formats across the site.

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Can we move the repo name to the top to the left of open in github button? We can keep icon only for open in github in mobile view.

Also, it is totally fine to remove the github org from the card in desktop view too if it is making things complicated to achieve it

image

File: app/releases/page.tsx (hunk 1 above)
Comment (rash-27) [reply app/releases/page.tsx:67]: okk I will make changes and update you , and I have used that format as it is also used in the Release section in homepage

Comment (rithviknishad) [reply app/releases/page.tsx:67]: nice find, update there too 😄

Comment (rash-27) [reply app/releases/page.tsx:67]: okk , ill update it

Comment (rash-27) [reply app/releases/page.tsx:67]: image
image

how does this look ? i have displayed only the git hub icon when dimension is less than 370px , and also I have one doubt like is it not necessary to put the version number as we do previously before shifting to octokit ?

Comment (rithviknishad) [reply app/releases/page.tsx:67]: Yes, it is important to show the version number.

Comment (rash-27) [reply app/releases/page.tsx:67]: Currently i guess it is not fetched , can i take up this issue and solve ?

Comment (rithviknishad) [reply app/releases/page.tsx:67]: yup

Comment (rash-27) [reply app/releases/page.tsx:67]: Should i solve that part too under this issue ?

Comment (rithviknishad) [reply app/releases/page.tsx:67]: Create a new issue, but could be solved in the same PR by linking multiple issues to it

Comment (rash-27) [reply app/releases/page.tsx:67]: okk

**Comment by rash-27** [comment]
is it ok ?
image

**Comment by rithviknishad** [comment]
Yup Nice! can we also add a tag icon before the version?

**Comment by rithviknishad** [comment]
Also, can we mark the latest release of a particular repo with the "latest" tag/chip?

**Comment by rash-27** [comment]
Yeah sure , I'll work on it

**Comment by rash-27** [comment]
@rithviknishad I couldnt find the latest tag in react-icons , so i have added a star as shown for latest release in a particular repository , how does this look ?
image

**Comment by rithviknishad** [comment]
This looks nice, but are we not showing the repository name? 👀

**Comment by rithviknishad** [comment]
I think, just specifying "Latest" would be better than trying to use an icon. A user may not recognize that this is the latest by looking at the star.

**Comment by rash-27** [comment]
In phone view I am only displaying the version , but in remaining both are displayed

**Comment by rithviknishad** [comment]
It is okay to hide org name in both views. But I don't think it is okay to hide the repository name. An organization may have multiple products/repositories each having its releases. One would not be able to distinguish which version belongs to which repository by looking at this from mobile right?

This would've been fine only if the releases page was specific to a repository.

**Comment by rash-27** [comment]
okk , then ill display just the repository name in phone view , is it sufficient ?
image
and here is the latest tag here ok ?

**Comment by rithviknishad** [comment]
Lets show both repo name and version name in mobile view. It's fine to make the text smaller or put things in newline in mobile view

**Comment by rash-27** [comment]
Ok , I'll update it

File: app/releases/page.tsx [hunk 2]
```diff
@@ -26,43 +31,52 @@ export default async function Page() {
...
+                    className="relative hidden pt-1 text-lg sm:inline sm:pt-1.5 "
+                    size={30}
+                    color="green"
+                  />
+                  {release.name}
+                  {myyRepos.has(release.repository) &&
+                    myyRepos.delete(release.repository) && (
+                      <span className="relative ml-2 rounded-full border border-green-700 p-1 pt-2.5 text-xs text-green-700 ">
+                        Latest
+                      </span>
```
Comment (rithviknishad) [inline app/releases/page.tsx:51]: the "latest" chip has improper padding. this does not look nice...

image


```suggestion
                      <span className="relative ml-2 rounded-full border border-green-700 p-1 text-xs text-green-700 ">
                        Latest
                      </span>
```

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**